vignettes/dependencies.Rmd

```diff
@@ -0,0 +1,91 @@
+---
+title: "Dependency resolution for R package development"
+author: "Jim Hester, Hadley Wickham, Gábor Csárdi"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    keep_md: true
+vignette: >
+  %\VignetteIndexEntry{Dependency resolution for R package development}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Package remotes
+
+Remotes, just like devtools, supports package dependency installation for packages not
+yet in a standard package repository such as [CRAN](https://cran.r-project.org)
+or [Bioconductor](http://bioconductor.org).
+
+You can mark any regular dependency defined in the `Depends`, `Imports`,
+`Suggests` or `Enhances` fields as being installed from a remote location by
+adding the remote location to `Remotes` in your `DESCRIPTION` file. This will
+cause remotes to download and install them from the specified location,
+instead of CRAN.
+
+The remote dependencies specified in `Remotes` should be described in the following form.
+
+```
+Remotes: [type::]<Repository>, [type2::]<Repository2>
+```
+
+The `type` is an optional parameter.  If the type is missing the default is
+to install from GitHub. Additional remote dependencies should be separated by
+commas, just like normal dependencies elsewhere in the `DESCRIPTION` file.
+
+### Github
+
+Because github is the most commonly used unofficial package distribution in R, it's the default:
+
+```yaml
+Remotes: r-lib/testthat
+```
+
+You can also specify a specific hash, tag, or pull request (using the same syntax as `install_github()` if you want a particular commit. Otherwise the latest commit on the master branch is used.
+
+```yaml
+Remotes: r-lib/httr@v0.4,
+  klutometis/roxygen#142,
+  r-lib/testthat@c67018fa4970
```

Comment:
Maybe we should mention the `@*release` syntax here as well? It is not very well known, I am not entirely positive it will even work with remotes actually...

Reply:
it used  to work, but yeah, it does not  work now:
```r
❯ install_github("r-lib/R6@*release")
Error in github_resolve_ref.github_release(meta$ref %||% ref, meta, auth_token) :
  No releases found for repo r-lib/R6.
```
https://github.com/r-lib/remotes/issues/200